### PR TITLE
[IMP] website_sale: display warning in notifications

### PR DESCRIPTION
**Pull request by vava-odoo**
When adding items to the cart, the server returns warnings that are never displayed to the customer. This commit makes use of the new notification system of website_sale to display it.

task-3502911

### CODE REVIEW

**Review by chevalierv — APPROVED** [review]
Hello @vava-odoo 

Thanks for your work on this! LGTM :+1: 

Have a nice day :sun_with_face:

File: addons/website_sale_stock/models/sale_order.py [hunk 1]
```diff
@@ -41,14 +41,26 @@ def _verify_updated_quantity(self, order_line, product_id, new_qty, **kwargs):
...
                 else:  # 0 or negative allowed_qty
                     # if existing line: it will be deleted
                     # if no existing line: no line will be created
-                    self.shop_warning = _(
-                        "Some products became unavailable and your cart has been updated. We're sorry for the inconvenience.")
-                return allowed_line_qty, order_line.shop_warning or self.shop_warning
+                    returned_warning = _(
+                        "The item has not been added to your cart since it is not available for the"
+                        " requested period"
+                    )
```
Comment (Feyensv) [inline addons/website_sale_stock/models/sale_order.py:51]: 'period' doesn't make any sense in standard stock logic (+ pot update if you make that change)

File: addons/website_sale_stock/models/sale_order.py [hunk 2]
```diff
@@ -41,14 +41,25 @@ def _verify_updated_quantity(self, order_line, product_id, new_qty, **kwargs):
...
         return super()._verify_updated_quantity(order_line, product_id, new_qty, **kwargs)
 
+    def _cart_update_order_line(self, product_id, quantity, order_line, **kwargs):
+        res = super()._cart_update_order_line(product_id, quantity, order_line, **kwargs)
+        if order_line and quantity <= 0:
+            self.shop_warning = _(
+                "Some products became unavailable and your cart has been updated. We're sorry for "
+                "the inconvenience."
+            )
+        return res
```
Comment (Feyensv) [inline addons/website_sale_stock/models/sale_order.py:61]: I don't really like this change, it's not restricted to the stock logic and might replace the warning in other situations depending on other potential overrides of `_verify_update_quantity`

File: addons/website_sale_stock/models/sale_order.py [hunk 3]
```diff
@@ -41,14 +41,25 @@ def _verify_updated_quantity(self, order_line, product_id, new_qty, **kwargs):
...
+                    returned_warning = order_line.shop_warning or self.shop_warning
                 else:  # 0 or negative allowed_qty
                     # if existing line: it will be deleted
                     # if no existing line: no line will be created
-                    self.shop_warning = _(
-                        "Some products became unavailable and your cart has been updated. We're sorry for the inconvenience.")
-                return allowed_line_qty, order_line.shop_warning or self.shop_warning
+                    returned_warning = _(
+                        "The item has not been added to your cart since it is not available."
+                    )
```
Comment (Feyensv) [inline addons/website_sale_stock/models/sale_order.py:50]:
```python
if order_line:
    self.shop_warning = _(...)
    returned_warning = self.shop_warning
else:
   returned_warning = _(...)
```

Can't you handle the two error messages here, without a new override of `_cart_update_order_line` ?

Comment (vava-odoo) [reply addons/website_sale_stock/models/sale_order.py]: Yes, good idea!

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**